components/Common/Pagination/useGetPageElements.tsx

```diff
@@ -0,0 +1,146 @@
+'use client';
+import Link from 'next/link';
+import { useIntl } from 'react-intl';
+
+import styles from './index.module.css';
+
+import type { PaginationProps } from '.';
+
+type ParsedPage = PaginationProps['pages'][number] & {
+  pageNumber: number;
+};
+
+function parsePages(pages: PaginationProps['pages']): ParsedPage[] {
+  return pages.map(({ url }, index) => ({
+    url,
+    pageNumber: index + 1,
+  }));
+}
+
+const ellipsis = (
+  <span aria-hidden="true" className={styles.ellipsis}>
+    ...
+  </span>
+);
+
+function createPageElements({
+  parsedPages,
+  currentPage,
+  intl,
+}: Pick<PaginationProps, 'currentPage'> & {
+  parsedPages: ParsedPage[];
+  intl: ReturnType<typeof useIntl>;
```

Comment:
Since intl is called just for one entry, I wonder if you couldn't just use `useIntl` with dependency injection as passing `intl` as a prop (which feels odd)

Reply:
Using `useIntl` directly within the function was not an option, as that would prevent me from using this function conditionally, due to the rules of hooks.

Could you explain how would you do it with dependency injection? Isn't that what the `intl` prop is doing already?

Also, using `withIntil` for a regular function did not come to my mind. I can try that.

Reply:
Yup, `withIntl` should work. As again, for all intends and purposes this is a component, but I would rather that you extract this to its own file and have its own tests separated.